cmd/manager/main.go

```diff
@@ -140,6 +156,43 @@ func main() {
 		}
 	}
 
+	if err := serveCRMetrics(cfg); err != nil {
+		log.Info("Could not generate and serve custom resource metrics", "error", err.Error())
+	}
+
+	// Add to the below struct any other metrics ports to expose.
+	servicePorts := []corev1.ServicePort{
+		{Port: metricsPort, Name: metrics.OperatorPortName, Protocol: corev1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: metricsPort}},
+		{Port: operatorMetricsPort, Name: metrics.CRPortName, Protocol: corev1.ProtocolTCP, TargetPort: intstr.IntOrString{Type: intstr.Int, IntVal: operatorMetricsPort}},
+	}
+
+	// Create Service object to expose the metrics port(s).
+	ctx := context.TODO()
+	service, err := metrics.CreateMetricsService(ctx, cfg, servicePorts)
+	if err != nil {
+		log.Info("Could not create metrics Service", "error", err.Error())
+	}
+
+	operatorNs, err := k8sutil.GetOperatorNamespace()
+	if err != nil {
+		fatal(err)
+	}
+
+	// CreateServiceMonitors will automatically create the prometheus-operator
+	// ServiceMonitor resources necessary to configure Prometheus to scrape
+	// metrics from this operator.
+	services := []*corev1.Service{service}
```

Comment:
Is it OK to call this with a nil `service`?

It looks like it'll wind up creating a `v1.Service` with `Ports` set to `ServicePort{nil}` - perhaps this block should be inside an else?

Reply:
I took this code from operator-sdk's default generated project code.
Yes, we can move it to an else of the above service creation error check block.